File: app/src/main/java/com/example/voice_talk_android/bridge/NavigationBarComponent.kt
package com.example.voice_talk_android.bridge

import android.util.Log
import android.view.View
import com.google.android.material.appbar.AppBarLayout
import dev.hotwire.core.bridge.BridgeComponent
import dev.hotwire.core.bridge.BridgeDelegate
import dev.hotwire.core.bridge.Message
import dev.hotwire.navigation.destinations.HotwireDestination
import kotlinx.serialization.Serializable

class NavigationBarComponent(
    name: String,
    private val delegate: BridgeDelegate<HotwireDestination>
) : BridgeComponent<HotwireDestination>(name, delegate) {

    companion object {
        private const val TAG = "NavigationBar"
    }

    override fun onReceive(message: Message) {
        Log.d(TAG, "📱 NavigationBarComponent received: ${message.event}")

        when (message.event) {
            "hide" -> handleHide(message)
            "show" -> handleShow(message)
            else -> Log.w(TAG, "❌ Unknown event: ${message.event}")
        }
    }

    // MARK: - Toolbar 숨기기

    private fun handleHide(message: Message) {
        Log.d(TAG, "📱 Hiding toolbar...")

        try {
            val fragment = delegate.destination.fragment
            Log.d(TAG, "  - Fragment: ${fragment::class.simpleName}")
            Log.d(TAG, "  - Fragment.view: ${fragment.view}")
            
            val appBar = fragment.view?.findViewById<AppBarLayout>(com.example.voice_talk_android.R.id.app_bar)
            Log.d(TAG, "  - AppBar found: ${appBar != null}")
            
            if (appBar != null) {
                Log.d(TAG, "  - AppBar current visibility: ${appBar.visibility}")
                appBar.visibility = View.GONE
                Log.d(TAG, "  - AppBar new visibility: ${appBar.visibility}")
                Log.d(TAG, "✅ Toolbar hidden via findViewById")
                replyTo(message.event, SuccessResponse(true))
            } else {
                Log.w(TAG, "⚠️ AppBar not found - fragment.view might be null or layout doesn't have app_bar")
                Log.w(TAG, "  - Trying to find all views in fragment...")
                fragment.view?.let { v ->
                    Log.w(TAG, "  - Fragment view ID: ${v.id}")
                    Log.w(TAG, "  - Fragment view class: ${v::class.simpleName}")
                }
                replyTo(message.event, SuccessResponse(false))
            }
        } catch (e: Exception) {
            Log.e(TAG, "❌ Failed to hide toolbar: ${e.message}")
            e.printStackTrace()
            replyTo(message.event, SuccessResponse(false))
        }
    }

    // MARK: - Toolbar 표시

    private fun handleShow(message: Message) {
        Log.d(TAG, "📱 Showing toolbar...")

        try {
            val fragment = delegate.destination.fragment
            val appBar = fragment.view?.findViewById<AppBarLayout>(com.example.voice_talk_android.R.id.app_bar)
            
            if (appBar != null) {
                appBar.visibility = View.VISIBLE
                Log.d(TAG, "✅ Toolbar shown via findViewById")
                replyTo(message.event, SuccessResponse(true))
            } else {
                Log.w(TAG, "⚠️ AppBar not found")
                replyTo(message.event, SuccessResponse(false))
            }
        } catch (e: Exception) {
            Log.e(TAG, "❌ Failed to show toolbar: ${e.message}")
            replyTo(message.event, SuccessResponse(false))
        }
    }

    // MARK: - Response Data Classes

    @Serializable
    data class SuccessResponse(val success: Boolean)
}

File: voice_talk_ios/Bridge/Components/NavigationBarComponent.swift
//
//  NavigationBarComponent.swift
//  voice_talk_ios
//
//  Navigation bar를 제어하는 Bridge Component
//

import Foundation
import HotwireNative
import UIKit

final class NavigationBarComponent: BridgeComponent {
    override class var name: String { "navigation-bar" }
    
    // MARK: - Message Handling
    
    override func onReceive(message: Message) {
        guard let event = Event(rawValue: message.event) else {
            return
        }
        
        switch event {
        case .hide:
            handleHide(message: message)
        case .show:
            handleShow(message: message)
        }
    }
    
    // MARK: - Private Methods
    
    private func handleHide(message: Message) {
        guard let viewController = delegate?.destination as? UIViewController else { return }
        viewController.navigationController?.setNavigationBarHidden(true, animated: true)
        reply(to: "hide")
        print("✅ Navigation bar hidden")
    }
    
    private func handleShow(message: Message) {
        guard let viewController = delegate?.destination as? UIViewController else { return }
        viewController.navigationController?.setNavigationBarHidden(false, animated: true)
        reply(to: "show")
        print("✅ Navigation bar shown")
    }
}

// MARK: - Event Types

private extension NavigationBarComponent {
    enum Event: String {
        case hide
        case show
    }
}

